gcg/entrypoint.py

```diff
@@ -197,34 +197,32 @@ def collate_entry_header_data(repo, entries, options):
     return retval
 
 
-def log_entry_header_from_commit(the_commit):
+def log_entry_header_from_commit(the_commit, options):
     """
     Retrieve information for the release line from the commit
     :param the_commit: a git.Commit object
     :return: dictionary
     """
-    # FIXME: how about timezones, dear people?
     hdr = dict()
-    hdr['date'] = time.gmtime(the_commit.authored_date)
-    hdr['date_rpm'] = time.strftime('%a %b %d %Y', hdr['date'])
-    hdr['date_deb'] = email.utils.formatdate(the_commit.authored_date)
+    hdr['date'] = datetime.fromtimestamp(the_commit.authored_date, options.tz)
+    hdr['date_rpm'] = hdr['date'].strftime('%a %b %d %Y')
+    hdr['date_deb'] = hdr['date'].strftime('%a, %d %b %Y %R:%S %z')
```

Comment:
What would be the rationale to swap the [existing format function](https://docs.python.org/2.7/library/email.utils.html#email.utils.formatdate) (which returns an RFC 2822 compatible format already)? I'm not opposed to the change, just curious :-)

I wasn't able to find the `%R` formatter: https://docs.python.org/3/library/datetime.html#strftime-strptime-behavior or https://docs.python.org/2.7/library/datetime.html#strftime-strptime-behavior though, where does it come from?

Reply:
> What would be the rationale to swap the [existing format function](https://docs.python.org/2.7/library/email.utils.html#email.utils.formatdate) (which returns an RFC 2822 compatible format already)? I'm not opposed to the change, just curious :-)
> 
> I wasn't able to find the `%R` formatter: https://docs.python.org/3/library/datetime.html#strftime-strptime-behavior or https://docs.python.org/2.7/library/datetime.html#strftime-strptime-behavior though, where does it come from?

The rationale is: time.localtime(timestamp) does not work properly (the reason why pytz et.al. exist in the first place) and therefore I had to use datetime.fromtimestamp().strftime() anyway to get the rpm date right. Given that, I saved an additional import using the same function for formatting deb dates.

Regarding the %R:
man strftime(3). Knowing for a fact, that strftime simply uses glibc's strftime. I should have read on, though: %T is does the same like %R:%S
Going to use %T as it is more common...

Reply:
Thanks, that's a good reason.

If I may suggest, it might be a good idea to use the formatting tokens from Python standard rather than glibc docs, so  just plain old `%H:%M:%S`.

One can't rely on every Python implementation to use glibc.